Client/Pages/Budget/EditBudgetLineItem.razor

```diff
@@ -0,0 +1,56 @@
+@page "/budgetlineitem/edit"
+@inject NavigationManager NavManager
+@using Cheddar.Client.ViewModels
+@inject BudgetLineItemViewModel BudgetLineViewModel
+@inject BudgetViewModel budget
+@inject BudgetSettingsViewModel BudgetSettingsViewModel
+@inject ApplicationState ApplicationState
+
+<PageTitle>New Budget Line Item</PageTitle>
+
+<div class="row px-3">
+    <div class="col-md-6 col-lg-4 p-3">
+        <RadzenCard>
+            <h4 class="mb-4">Budget Line Item Name</h4>
+            <RadzenTextBox Placeholder="Apple Music" @bind-Value="@ApplicationState.BudgetLineItemModel.BudgetLineName" />
+        </RadzenCard>
+    </div>
+    <div class="col-md-6 col-lg-4 p-3">
+        <RadzenCard>
+            <h4 class="mb-4">Monthly Cost</h4>
+            <RadzenNumeric TValue="double" Placeholder="14.99" @bind-Value="@ApplicationState.BudgetLineItemModel.Cost" />
+        </RadzenCard>
+    </div>
+    <div class="col-md-6 col-lg-4 col-xl-3 p-3">
+        <RadzenCard>
+            <h4 class="mb-4">Category</h4>
+            <RadzenDropDown Class="w-100" 
+                            Data=@BudgetLineViewModel.budgetCategories TValue="BudgetCategoriesModel" TextProperty="Name" @bind-Value="@ApplicationState.BudgetLineItemModel.Category"/>
+        </RadzenCard>
+    </div>
+    <div class="col-md-6 col-lg-4 col-xl-3 p-3">
+        <RadzenCard>
+            <h4 class="mb-4">Payment Methods</h4>
+            <RadzenDropDown AllowClear="true" Class="w-100" 
+                            Data=@BudgetLineViewModel.paymentMethods TValue="PaymentMethodsModel" TextProperty="Name" @bind-Value="@ApplicationState.BudgetLineItemModel.PaymentMethod"/>
+        </RadzenCard>
+    </div>
+    <div class="col-lg-6 col-xl-4 p-3">
+        <RadzenCard>
+            <h4 class="mb-4">Contract End Date</h4>
+            <RadzenDatePicker DateFormat="d" @bind-Value="@ApplicationState.BudgetLineItemModel.ContractEndDate" />
+        </RadzenCard>
+    </div>
+    <RadzenButton class="col-xl-3" Click="@(() => BudgetLineViewModel.UpdateBudgetLineItem(ApplicationState.BudgetLineItemModel))" Text="Save" ButtonStyle="ButtonStyle.Primary"></RadzenButton>
+    <RadzenButton class="col-xl-3" Click="@(() => BudgetLineViewModel.DeleteBudgetLineItem(ApplicationState.BudgetLineItemModel))" Text="Delete" ButtonStyle="ButtonStyle.Primary"></RadzenButton>
+</div>
+
+
+@code {
+
+    protected override async Task OnInitializedAsync() {
```

Comment:
Should handle if the app state isn't set (`null`)

Reply:
I wasn't sure how best to handle it so I declared a new model so it doesn't throw errors